client/app/cloud/project/billing/cloud-project-billing-legacy.service.js

```diff
@@ -0,0 +1,297 @@
+        0,
+      );
+
+      return ({
+        monthlyInstances,
+        total: roundNumber(monthlyTotal, 2),
+      });
+    }
+
+    function initObjectStorageList(consumptionDetails) {
+      if (!consumptionDetails.hourlyBilling || !consumptionDetails.hourlyBilling.hourlyUsage) {
+        return getEmptyConsumption('objectStorages');
+      }
+
+      const objectStorages = _.reject(
+        _.map(
+          consumptionDetails.hourlyBilling.hourlyUsage.storage,
+          objectStorage => _.set(objectStorage, 'totalPrice', roundNumber(objectStorage.totalPrice, 2)),
+        ), { type: 'pca' },
+      );
+
+      const objectStorageTotal = _.reduce(
+        objectStorages,
+        (sum, storage) => sum + roundNumber(storage.totalPrice, 2),
+        0,
+      );
+
+      return ({
+        objectStorages,
+        total: roundNumber(objectStorageTotal, 2),
+      });
+    }
+
+    function initArchiveStorageList(consumptionDetails) {
+      if (!consumptionDetails.hourlyBilling || !consumptionDetails.hourlyBilling.hourlyUsage) {
+        return getEmptyConsumption('archiveStorages');
+      }
+
+      const archiveStorages = _.filter(
+        _.map(
+          consumptionDetails.hourlyBilling.hourlyUsage.storage,
+          objectStorage => _.set(objectStorage, 'totalPrice', roundNumber(objectStorage.totalPrice, 2)),
+        ), { type: 'pca' },
+      );
+      const archiveStorageTotal = _.reduce(
+        archiveStorages,
+        (sum, archiveStorage) => sum + roundNumber(archiveStorage.totalPrice, 2),
+        0,
+      );
+
+      return ({
+        archiveStorages,
+        total: roundNumber(archiveStorageTotal, 2),
+      });
+    }
+
+    function initSnapshotList(consumptionDetails) {
+      if (!consumptionDetails.hourlyBilling || !consumptionDetails.hourlyBilling.hourlyUsage) {
+        return getEmptyConsumption('snapshots');
+      }
+
+      const snapshots = _.map(
+        consumptionDetails.hourlyBilling.hourlyUsage.snapshot,
+        snapshot => _.set(snapshot, 'totalPrice', roundNumber(snapshot.totalPrice, 2)),
+      );
+
+      const snapshotTotal = _.reduce(
+        consumptionDetails.hourlyBilling.hourlyUsage.snapshot,
+        (sum, snapshot) => sum + roundNumber(snapshot.totalPrice, 2),
+        0,
+      );
+
+      return ({
+        snapshots,
+        total: roundNumber(snapshotTotal, 2),
+      });
+    }
+
+    function initVolumeList(consumptionDetails) {
+      if (!consumptionDetails.hourlyBilling || !consumptionDetails.hourlyBilling.hourlyUsage) {
+        return getEmptyConsumption('volumes');
+      }
+      const volumes = _.flatten(_.map(
+        consumptionDetails.hourlyBilling.hourlyUsage.volume,
+        volume => _.map(volume.details, (detail) => {
+          const newDetail = _.clone(detail);
+          newDetail.totalPrice = roundNumber(newDetail.totalPrice, 2);
+          return _.extend(newDetail, { type: volume.type, region: volume.region });
```

Comment:
```suggestion
          return {
              ...detail,
              totalPrice: roundNumber(detail.totalPrice,  2),
              type: volume.type,
              region: volume.region,
          };
```

Reply:
I've just change the service to be stateless, this will be removed once migration is completed